IBX-4091: Added predictive and recurring purchase models (#177)
ezsystems/user-documentation: docs/personalization/recommendation_models.md

usage history (recent events are more important) and category-based filtering (bestsellers 
in the selected category and/or subcategories).

#### Predictive

> Predicts user's purchase behavior and trends, and recommends items based on this behavior.

Reviewer: users'
In case of predictive model, all purchases or clicks of particular item are taken into account

> Predicts users' purchase behavior and trends, and recommends items based on this behavior.

Reviewer 2: ```suggestion
A predictive popularity model predicts users' purchase behavior and trends, and recommends items based on this behavior.
```

> The model analyzes trends within a configured time period (for example, within a 30-day timeframe), and predicts that a new item may have the same trend, and displays the predicted item in recommendations.

Reviewer 2: ```suggestion
The model analyzes trends within a configured time period (for example, within a 30-day timeframe), predicts that a new item may have the same trend, and displays the predicted item in recommendations.
```

Pseudo recommendation model that shows the user products from their own history. 
For example, the "You have just watched" box.

#### Recurring purchase

> Recognizes purchase patterns and returns recommendations for items where these pattern can occur.

Reviewer 2: ```suggestion
A recurring purchase model recognizes purchase patterns and returns recommendations for items where these patterns can occur.
```

This model is based on predictable purchase occuring at regular intervals going forward with a relatively high degree of certainty.

> Model recommends items which are suitable for this pattern. Item recommendations are displayed 20% before the estimated pattern of repeating purchasing the product.

Reviewer 2: ```suggestion
The model recommends items which are suitable for this pattern. Item recommendations are displayed 20% before the estimated pattern of repeating purchasing the product.
```
"are displayed 20% before" - that's somehow awkward, even with the explanation below

> It means, for example, if the pattern covers 100 days, when the optimal time comes from the day 80. the recommendation begins to display. When user purchases the recommended item on the day 92., 

Reviewer 2: ```suggestion
It means, for example, that if the pattern covers 100 days, when the optimal time comes from day 80., the recommendation begins to display. When a user purchases the recommended item on day 92., 
```

> the recommendation is no longer showed and the counter resets.

Reviewer 2: ```suggestion
the recommendation is no longer shown and the counter resets.
```

!!! tip

>     Minimum time period for this model is 4 days, but the model works better if you set longer time period.

Reviewer 2: ```suggestion
    Minimum time period for this model is 4 days, but the model works better if you set a longer time period.
    
```

Pseudo recommendation model that shows the user products from their own history. 
For example, the "You have just watched" box.

#### Recurring purchase

A recurring purchase model recognizes purchase patterns and returns recommendations for items where these patterns can occur.
This model is based on predictable purchase occuring at regular intervals going forward with a relatively high degree of certainty.

> The model recommends items which are suitable for this pattern. Item recommendations are displayed 20% before the estimated pattern of repeating purchasing the product.

Author: ```suggestion
The model recommends items which are suitable for this pattern. Item recommendations are displayed 20% before the completion of estimated pattern of repeating purchasing the product.
```
